compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs

```diff
@@ -117,16 +117,22 @@ onDecl o = descendBi onExp o : extraDecls o
             where ctors = dd_cons $ tcdDataDefn x
         extraDecls _ = []
 
-getFields :: TyClDecl GhcPs -> [(LHsType GhcPs, IdP GhcPs, FieldOcc GhcPs, HsType GhcPs)]
+-- Process a Haskell data type definition. Walk its constructors. For
+-- each, harvest a tuple of the parent type, the ctor name, the field
+-- name and the field type.
+getFields :: TyClDecl GhcPs ->
+               [(LHsType GhcPs, IdP GhcPs, FieldOcc GhcPs, HsType GhcPs)]
 getFields DataDecl{tcdDataDefn=HsDataDefn{..}, ..} = concatMap ctor dd_cons
     where
         ctor (L _ ConDeclH98{con_args=RecCon (L _ fields),con_name=L _ name}) = concatMap (field name) fields
-        ctor _ = []
+        ctor _ = [] -- GADT ctors ignored.
```

Comment:
Leverage the compiler to "check" the comment:
```suggestion
        ctor (L _ ConDeclGADT{}) = []
```

Reply:
✔️